.github/workflows/ci-tests.yml

```diff
@@ -358,14 +358,84 @@ jobs:
           name: coverage-data
 
       - name: Combine coverage & fail if it's <100%.
+        id: report_coverage
+        # NOTE: Setting the pipefail option here means that even when
+        # piping the output to `tee`, we still get the exit code of the
+        # `coverage report` command.
         run: |
+          set -o pipefail
           python -m coverage combine
           python -m coverage html --skip-covered --skip-empty
-          python -m coverage report --fail-under=100
+          python -m coverage report --fail-under=100 --skip-covered --skip-empty -m | tee coverage-report.txt
 
       - name: Upload HTML report if check failed.
         uses: actions/upload-artifact@v3
         with:
           name: html-report
           path: htmlcov
         if: ${{ failure() }}
+
+      - name: Update PR Comment.
+        uses: actions/github-script@v6
+        # Run even if the coverage step failed.
+        if: always()
+        with:
+          script: |
+            // First list the existing comments
+
+            const trigger_str = 'Coverage Results';
+            console.log("Getting existing comments...");
+
+            const issue_number = context.issue.number;
+            console.log("Issue number: " + issue_number);
```

Comment:
@alanmcruickshank it's failing because
`Issue number: undefined`

Reply:
Here's one suggestion, but I feel it wouldn't necessarily pick the right issue number: https://github.com/actions/github-script/issues/203#issuecomment-1027833024

Reply:
We should probably do this? https://github.com/actions/github-script/issues/203#issuecomment-1034195974